storage/migrations/0009_oasis_3_emerald_init.up.sql

```diff
@@ -16,10 +19,56 @@ CREATE TABLE IF NOT EXISTS oasis_3.emerald_rounds
   messages_hash    TEXT NOT NULL,
   in_messages_hash TEXT NOT NULL,
 
+  -- TODO: These should also be NOT NULL, but they're populated separately.
+  num_transactions INTEGER,
+  gas_used         BIGINT,
+  size             INTEGER,
+
   -- Arbitrary additional data.
   extra_data JSON
 );
 
+CREATE TABLE IF NOT EXISTS oasis_3.emerald_transactions
+(
+  round BIGINT NOT NULL,
+  tx_index INTEGER NOT NULL,
+  tx_hash TEXT NOT NULL,
+  tx_eth_hash TEXT,
+  PRIMARY KEY (round, tx_index)
```

Comment:
... or else the full raw tx. IMO we'll need to display at least some of these fields in the wallet and/or explorer later, and it's easier to parse everything now than do piecemeal update the APIs throughout the stack (to include the new field(s)) later. But I'll leave the judgement call to you.

Reply:
oh I forgot to reply here

https://docs.google.com/document/d/1moqmGcHosdnVUHfYT6M4mpQbWsWEnmHIOpFAfcosZdY/edit#heading=h.eetpelxkhnjv

parsing here would possibly be easier so that we take that parsed data with us across hard forks, using one tool: migrations. otherwise we'd have to do migrations _and_ write routines to parse earlier transaction versions.

however, to move over to the 'parse entirely while scanning' design is a larger change that we should scope out of this PR for getting the schemas merged

Reply:
What I was trying to say in my second comment in this thread was that I'm fine with not doing the "parse entirely while scanning" approach, but that in that case, we should probably include the full raw tx as a column.

Reply:
ok let's add it